src/core/symbolic.cc

```diff
@@ -10,7 +10,7 @@
 namespace nnvm {
 
 namespace symbol_constants {
-const char *kNamespaceSeparator = "_";
+const char *kNamespaceSeparator = "$";
```

Comment:
If we really want compatibility we can just replace $ with _ in frontend (the reverse is not possible)
but I think using _ as separation is really a bad idea and should be banned